app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,45 @@
+module SessionsHelper
+  # Logs in the given user.
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  # Returns the current logged-in user (if any).
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
+  end
+
+  # Returns true if the user is logged in, false otherwise.
+  def logged_in?
+    !current_user.nil?
```

Comment:
Asking if something is "not nil?" is a double negative, and difficult to reason about.

Rather do:

```suggestion
    current_user.present?
```

Reply:
https://devdocs.io/rails~4.2/object#method-i-present-3F